geocoder/geocoder.cpp

```diff
@@ -172,39 +205,92 @@ void Geocoder::Go(Context & ctx, Type type) const
 
       subquery.push_back(ctx.GetToken(j));
 
-      auto const * entries = m_hierarchy.GetEntries(subquery);
-      if (!entries || entries->empty())
-        continue;
-
       Layer curLayer;
       curLayer.m_type = type;
-      for (auto const & e : *entries)
-      {
-        if (e.m_type != type)
-          continue;
 
-        if (ctx.GetLayers().empty() || FindParent(ctx.GetLayers(), e))
-          curLayer.m_entries.emplace_back(&e);
+      // Buildings are indexed separately.
+      if (type == Type::Building)
+      {
+        FillBuildingsLayer(ctx, subquery, curLayer);
       }
-
-      if (!curLayer.m_entries.empty())
+      else
       {
-        ScopedMarkTokens mark(ctx, type, i, j + 1);
+        FillRegularLayer(ctx, type, subquery, curLayer);
+      }
+
+      if (curLayer.m_entries.empty())
+        continue;
 
-        double const certainty =
-            static_cast<double>(ctx.GetNumUsedTokens()) / static_cast<double>(ctx.GetNumTokens());
+      ScopedMarkTokens mark(ctx, type, i, j + 1);
 
-        for (auto const * e : curLayer.m_entries)
-          ctx.AddResult(e->m_osmId, certainty);
+      // double const certainty =
+      //     static_cast<double>(ctx.GetNumUsedTokens()) /
```

Comment:
Why this code is commented?

Reply:
See the comment about certainty below.
I've removed the commented code for now.